system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.cpp

```diff
@@ -78,6 +78,10 @@ bool PeriodicMeasurementNode::setupStop()
     measurement_timer_->cancel();
     measurement_timer_.reset();
   }
+  if (publish_timer_) {
```

Comment:
why do we check, is there cases where this won't be true?

Reply:
This check has been made into an `assert()`.